### PR TITLE
[cloud-provider-gcp] Add cloud data discoverer

### PR DESCRIPTION
**Pull request by name212**
## Description
Add cloud data discoverer service.

## Why do we need it, and what problem does it solve?
Discovery instance types for cluster autoscaler.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-gcp
type: feature
summary: Add cloud data discoverer service which get information about available instance types for node groups.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/gcp'...

**Comment by name212** [comment]
e2e was passed have problem with cleanup

File: modules/030-cloud-provider-gcp/images/cloud-data-discoverer/main.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		Required().
+		StringVar(&credsFile)
+
+	kpApp.Flag("zone", "Zones for discovery").
+		Envar("GCP_ZONES").
+		Required().
+		StringsVar(&zones)
+
+	kpApp.Flag("project", "Zones for discovery").
+		Envar("GCP_ZONES").
```
Comment (yalosev) [inline modules/030-cloud-provider-gcp/images/cloud-data-discoverer/main.go:47]: Wrong comment and env var. It should contain something like `GCP_PROJECT`

**Review by yalosev — CHANGES_REQUESTED** [review]

File: modules/030-cloud-provider-gcp/images/cloud-data-discoverer/main.go (hunk 1 above)
Comment (name212) [reply modules/030-cloud-provider-gcp/images/cloud-data-discoverer/main.go]: Done

**Review by yalosev — APPROVED** [review]

**Pull request merged**